src/System.Private.CoreLib/shared/System/Memory.cs

```diff
@@ -104,8 +104,13 @@ public Memory(T[] array, int start, int length)
             }
             if (default(T) == null && array.GetType() != typeof(T[]))
                 ThrowHelper.ThrowArrayTypeMismatchException();
+#if BIT64
+            if ((ulong)(uint)start + (ulong)(uint)length > (ulong)(uint)array.Length)
```

Comment:
Can you please add a comment explaining why this works?

Reply:
Added in latest iteration.